docs/source/notes/autograd.rst

```diff
@@ -598,3 +643,83 @@ chain rule:
 
         .. math::
             \frac{\partial L}{\partial z^*} = 2 * Re(grad\_out^* * \frac{\partial s}{\partial z^{*}})
+
+.. _saved-tensors-hooks-doc:
+
+Hooks for saved tensors
+-----------------------
+
+You can control :ref:`how saved tensors are packed / unpacked
+<saved-tensors-doc>` by defining a pair of ``pack_hook`` / ``unpack_hook`` hooks.
+The ``pack_hook`` function should take a tensor as its single argument but can
+return any python object (e.g. another tensor, a tuple, or even a filename).
+The ``unpack_hook`` function takes as its single argument the output of
+``pack_hook`` and should return the original tensor to be used in the backward
+pass.
+
+An example of such pair is:
+
+.. code::
+
+    def pack_hook(x):
+        name = os.path.join(tmp_dir, str(uuid.uuid4()))
```

Comment:
I like this example, but we should use it to show how to handle the lifetime as well I think.
In particular, with this code, you're gonna "leak" everything you write on disk.

I think you want to show that the file on disk should live as long as the object your returned. And that the unpack must NOT delete the file as it might be called multiple times.

Reply:
The solution I have in mind is to clear the tmp_dir after each backward. Is there a better way you were thinking?

Reply:
You can have a file handle here that will delete the file when the handle is deleted. You can check the python's TempFile, I think they do that.